src/routers/SwitchRouter.js

```diff
@@ -88,35 +88,56 @@ export default (routeConfigs, config = {}) => {
     };
   }
 
-  function getNextState(prevState, possibleNextState) {
-    if (!prevState) {
-      return possibleNextState;
+  function getNextState(action, prevState, possibleNextState) {
```

Comment:
@brentvatne I added action to the params list there as it's needed for updateNextStateHistory.

Didn't want to move getNextState inside getStateForActions. Don't really like to have a lot of deeply nested closures whose captured variables are not very "visible", keeping the function outside makes it clearer which are the variable dependencies.

Also put updateNextStateHistory inside getNextState as it's the only place where it's used